shared/cluster.go

```diff
@@ -336,36 +336,49 @@ func GetPods(print bool) ([]Pod, error) {
 	return pods, nil
 }
 
+// GetPodsByNamespaceAndLabel returns pods parsed from kubectl get pods in a specific namespace with a specific label
+func GetPodsByNamespaceAndLabel(namespace, label string, print bool) ([]Pod, error) {
+	cmd := fmt.Sprintf("kubectl get pods -o wide --no-headers -n %s -l %s --kubeconfig=%s", namespace, label, KubeConfigFile)
+	res, err := RunCommandHost(cmd)
+	if err != nil {
+		return nil, ReturnLogError("failed to get pods: %w\n", err)
+	}
+
+	pods := parsePods(res)
+	if print {
+		fmt.Println(res)
+	}
+
+	return pods, nil
+}
+
 // parsePods parses the pods from the kubeclt get pods command.
 func parsePods(res string) []Pod {
 	pods := make([]Pod, 0, 10)
 	podList := strings.Split(strings.TrimSpace(res), "\n")
 
 	for _, rec := range podList {
+		offset := 0
 		fields := strings.Fields(rec)
-		if strings.TrimSpace(rec) == "" {
+		if strings.TrimSpace(rec) == "" || len(fields) < 9 {
 			continue
 		}
-
-		if len(fields) < 9 {
-			continue
+		var p Pod
+		if len(fields) == 10 {
```

Comment:
is this working fine ?

Reply:
Yep! 
<img width="655" alt="image" src="https://github.com/rancher/distros-test-framework/assets/55997940/5960e3d4-946f-4327-b6de-5b4a48220c06">


Length needs to be 10 with the Namespace because it's one additional field

Reply:
but its not 7 the pod structure ?

Reply:
Thanks, didn't see this message! What do you mean by "7 the pod structure"? This works in both cases the function is called: either with namespaces (kubectl get pods -A) or without (kubectl get pods -n default)

Reply:
no worries, sorry for dont send complete context , why are you doing `		if len(fields) == 10 {`  if the size is 8 ? with the namespace
 
```

type Pod struct {
	NameSpace string
	Name      string
	Ready     string
	Status    string
	Restarts  string
	Age       string
	NodeIP    string
	Node      string
}
```

Reply:
Ah the length is 10! There are two more fields that we don't have accounted for right now:
`NOMINATED NODE` and `READINESS GATES`. But those values are always <none> for us so we don't necessarily need to account for them right now.
```
NAMESPACE        NAME                                         READY   STATUS      RESTARTS   AGE     IP           NODE              NOMINATED NODE   READINESS GATES
kube-system      pod/coredns-77ccd57875-6mhdm                 1/1     Running     0          8m59s   10.42.0.6    ip-172-31-5-154   <none>           <none>
kube-system      pod/helm-install-traefik-crd-9kjss           0/1     Completed   0          8m59s   10.42.0.5    ip-172-31-5-154   <none>           <none>
kube-system      pod/helm-install-traefik-j6gws               0/1     Completed   1          8m59s   10.42.0.3    ip-172-31-5-154   <none>           <none>
kube-system      pod/local-path-provisioner-957fdf8bc-gnn45   1/1     Running     0          8m59s   10.42.0.4    ip-172-31-5-154   <none>           <none>
kube-system      pod/metrics-server-648b5df564-vjdjl          1/1     Running     0          8m59s   10.42.0.2    ip-172-31-5-154   <none>           <none>
kube-system      pod/svclb-traefik-6186304f-fpxmn             2/2     Running     0          8m24s   10.42.0.7    ip-172-31-5-154   <none>           <none>
kube-system      pod/traefik-768bdcdcdd-r2d7v                 1/1     Running     0          8m24s   10.42.0.8    ip-172-31-5-154   <none>           <none>
```

Reply:
yep , but we mapped this to 
```
type Pod struct {
	NameSpace string
	Name      string
	Ready     string
	Status    string
	Restarts  string
	Age       string
	NodeIP    string
	Node      string
}
```

- so our map ( not the real fields ) is 8, so lets count on the implementation details.

Reply:
if we need to grow to the real fields , then we validate against that.  We should always validate against whats implemented at the moment.

Reply:
either way its fine!

Reply:
Alright let's just leave as is then and we can add some enhancement in the future if needed to include the two additional fields in the map?

Reply:
leave as it is you mean ? 
```
if len(fields) == 8 {
```

if so yes, if its == 10 no. 

as i stated above we should validate , count and deal with what we have implemented.

and right now we dont have 10 implemented.

Reply:
`if len(fields) == 8 {`isn't allowed as that isn't a valid scenario and it will never hit that block, since the length of the fields is either 9 or 10. This checks the length of the _actual_ fields on the result, not the way we've mapped the fields.

Reply:
if thats the case, then yes. lets add as it is

Reply:
Updated!